### PR TITLE
Fix out of order exemplar error for native histograms

### PR DESCRIPTION
**Pull request by krajorama**
#### What this PR does

When receiving multiple exemplars for a native histogram in Mimir via remote write, sort them and only report an error if all are older than the latest exemplar as this could be a partial update.

#### Which issue(s) this PR fixes or relates to

Port of https://github.com/prometheus/prometheus/pull/13021

#### Checklist

- [X] Tests updated.
- N/A Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- N/A [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 * [BUGFIX] Ingester: Prevent timely compaction of empty blocks. #7624
 * [BUGFIX] querier: Don't cache context.Canceled errors for bucket index. #7620
 * [BUGFIX] Store-gateway: account for `"other"` time in LabelValues and LabelNames requests. #7622
+* [BUGFIX] Ingester: when receiving multiple exemplars for a native histogram in mimir via remote write, only report an error if all are older than the latest exemplar as this could be a partial update. #7640
```
Comment (beorn7) [inline CHANGELOG.md:14]: mimir → Mimir ?

**Review by beorn7 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (krajorama) [reply CHANGELOG.md:14]: oh, don't actually have to say Mimir at all, thanks

**Review by beorn7 — APPROVED** [review]

File: pkg/mimirpb/compat.go [hunk 2]
```diff
@@ -88,6 +88,13 @@ func (req *WriteRequest) AddHistogramSeries(lbls [][]LabelAdapter, histograms []
 	return req
 }
```
[the remark this answers is not in the document]
Comment (krajorama) [reply pkg/mimirpb/compat.go:90]: added

**Comment by krajorama** [comment]
For the record: I did test doing the sort in the distributor, however that would trigger a remarshalling of the remote write request and that's probably more costly than doing the sort in place.

**Pull request merged**